providers/fips/fipsprov.c

```diff
@@ -97,6 +99,9 @@ typedef struct fips_global_st {
 #include "fips_indicator_params.inc"
 #undef OSSL_FIPS_PARAM
 
+    /* Guards access to ondemand self-test */
+    CRYPTO_RWLOCK *ondemand_lock;
+    CRYPTO_THREAD_LOCAL ondemand_key;
```

Comment:
Do we want this per libctx?  That seems a little excessive.
It's also not consistent with the single global flag for the one deferred algorithm.

Reply:
I tried to have them as global static initially but all the infrastructure around RUN_ONCE() prevents their usage in FIPS_MODULE, so for now I stored them on the context, if we think we can relax the code that blocks those initializers for the fips provider this stuff can be very easily moved off again.

Reply:
you can use the CRYPTO_THREAD_run_once function directly in the FIPS module, rather than the RUN_ONCE macros.  I had to do the same thing for the CRYPTO_THREAD_[set|get]_local_ex apis.

Reply:
Alright, if this is allowed I can change it that way!